include/openthread/icmp6.h

```diff
@@ -76,8 +76,9 @@ typedef enum otIcmp6Type
  */
 typedef enum otIcmp6Code
 {
-    OT_ICMP6_CODE_DST_UNREACH_NO_ROUTE = 0, ///< Destination Unreachable No Route
-    OT_ICMP6_CODE_FRAGM_REAS_TIME_EX   = 1, ///< Fragment Reassembly Time Exceeded
+    OT_ICMP6_CODE_DST_UNREACH_NO_ROUTE   = 0, ///< Destination Unreachable - No Route
```

Comment:
Duplicate numbers are slightly confusing - could clarify like below, by saying to which ICMPv6 Type the Code belongs:

```suggestion
    OT_ICMP6_CODE_DST_UNREACH_NO_ROUTE   = 0, ///< Destination Unreachable - No Route (use only with Type 1)
```

Reply:
I opted to include associated `"Type"` in the comment before `"Code"` separated by a `-`, i.e. 
- `"Destination Unreachable (Type 1) - No Route"`
- `"Destination Unreachable (Type 1) - Administratively Prohibited"`
- `"Time Exceeded (Type 3) - Fragment Reassembly"`